src/Compilers/CSharp/Test/Syntax/Diagnostics/DiagnosticTest.cs

```diff
@@ -2150,6 +2150,124 @@ static void Main(string[] args)
+        /// </summary>
+        [Fact]
+        public void PossibleBadNegCastFixed()
+        {
+            var source = @"class Program
+{
+    static void Main()
+    {
+        var z = (System.ConsoleColor) (- 1);
+    }
+}";
+
+            var compilation = CreateCompilationWithMscorlib(source);
+            compilation.VerifyDiagnostics(new[]
+            {
+                // (5,13): warning CS0219: The variable 'z' is assigned but its value is never used
+                //         var z = (System.ConsoleColor) (- 1);
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "z").WithArguments("z").WithLocation(5, 13)
+            });
+        }
+
+        /// <summary>
+        ///    Tests if CS0075 - "To cast a negative value, you must enclose the value in parentheses" is only emitted if the left side is (would be) a cast expression.
+        /// </summary>
+        [Fact]
+        public void PossibleBadNegCastNotEmitted()
+        {
+            var source = @"class Program
+{
+    static void Main()
+    {
+        var y = ((System.ConsoleColor)) - 1;
+        var z = System.ConsoleColor - 1;
+    }
+}";
+
+            var compilation = CreateCompilationWithMscorlib(source);
+            compilation.VerifyDiagnostics(new[]
+            {
+                // (5,19): error CS0119: 'ConsoleColor' is a type, which is not valid in the given context
+                //         var x = ((System.ConsoleColor)) - 1;
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "System.ConsoleColor").WithArguments("System.ConsoleColor", "type").WithLocation(5, 19),
+                // (5,19): error CS0119: 'ConsoleColor' is a type, which is not valid in the given context
+                //         var x = ((System.ConsoleColor)) - 1;
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "System.ConsoleColor").WithArguments("System.ConsoleColor", "type").WithLocation(5, 19),
+                // (5,19): error CS0119: 'ConsoleColor' is a type, which is not valid in the given context
+                //         var x = ((System.ConsoleColor)) - 1;
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "System.ConsoleColor").WithArguments("System.ConsoleColor", "type").WithLocation(5, 19),
+                // (6,17): error CS0119: 'ConsoleColor' is a type, which is not valid in the given context
+                //         var y = System.ConsoleColor - 1;
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "System.ConsoleColor").WithArguments("System.ConsoleColor", "type").WithLocation(6, 17),
+            });
+        }
+
+        /// <summary>
+        ///    Tests if CS0075 - "To cast a negative value, you must enclose the value in parentheses" is also emitted for dynamic casts.
+        /// </summary>
+        [Fact]
+        public void PossibleBadNegCastDynamic()
+        {
+            var source = @"class Program
+{
+    static void Main()
+    {
+        var y = (dynamic) - 1;
```

Comment:
>var y = (dynamic) - 1; [](start = 8, length = 22)

Please also test this scenario when :
- there is a local named "dynamic" in scope.
- there is a method named "dynamic" in scope.